public/app/plugins/datasource/cloudwatch/ConfigEditor.tsx

```diff
@@ -0,0 +1,389 @@
+import React, { PureComponent, ChangeEvent } from 'react';
+import { DataSourcePluginOptionsEditorProps, FormLabel, Select, Input, Button } from '@grafana/ui';
+import { SelectableValue } from '@grafana/data';
+import DatasourceSrv, { getDatasourceSrv } from 'app/features/plugins/datasource_srv';
+import CloudWatchDatasource from './datasource';
+
+export type Props = DataSourcePluginOptionsEditorProps<any>;
+
+export interface State {
+  config: any;
+  authProviderOptions: SelectableValue[];
+  regions: SelectableValue[];
+}
+
+export class ConfigEditor extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    const { options } = this.props;
+
+    this.state = {
+      config: ConfigEditor.keyFill(options),
+      authProviderOptions: [
+        { label: 'Access & secret key', value: 'keys' },
+        { label: 'Credentials file', value: 'credentials' },
+        { label: 'ARN', value: 'arn' },
+      ],
+      regions: [],
+    };
+
+    this.datasourceSrv = getDatasourceSrv();
+
+    this.updateDatasource(this.state.config);
+
+    this.getRegions();
```

Comment:
calling async functions that set state should be in componentDidMount and not in the constructor